src/view/adminhtml/web/js/datepicker.js

```diff
@@ -0,0 +1,22 @@
+//This script fixes incorrect positioning in modal datepickers.
+//This file should be removed after magento fixes the datepickers.
+
+require(["jquery"], function ($) {
+    $(window).on("load", function () {
+        if (window.location.pathname.includes("/admin/catalog/product/edit/id")) {
```

Comment:
This will fix the datepicker on modals on the product edit page.
But this will not fix the datepicker on modal on other pages.
Please remove this check.